### ISSUE TITLE
Renovate updating wrong dependency with artifact version that doesn't exist

### ISSUE THREAD

**Issue by Elm0xz** [renovatebot/renovate#18937]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.240.5

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.4.3-ee

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

When updating `jackson` version, Renovate also tried to update `jackson-databind` version (which wasn't mentioned as a dependency to be updated in that MR) to one that does not exist in maven repository. 

### Relevant debug logs

<details><summary>Logs</summary>

**Information about dependencies to be updated:**

This MR contains the following updates:

| Package | Update | Change |
|---|---|---|
| [com.fasterxml.jackson.dataformat:jackson-dataformat-cbor](https://nexus.ocado.tech/service/rest/repository/browse/maven-central/com/fasterxml/jackson/dataformat) | minor | `2.13.4` -> `2.14.0` |
| [com.fasterxml.jackson.dataformat:jackson-dataformat-yaml](https://nexus.ocado.tech/service/rest/repository/browse/maven-central/com/fasterxml/jackson/dataformat) | minor | `2.13.4` -> `2.14.0` |
| [com.fasterxml.jackson.datatype:jackson-datatype-jdk8](https://nexus.ocado.tech/service/rest/repository/browse/maven-central/com/fasterxml/jackson/datatype) | minor | `2.13.4` -> `2.14.0` |
| [com.fasterxml.jackson.datatype:jackson-datatype-jsr310](https://nexus.ocado.tech/service/rest/repository/browse/maven-central/com/fasterxml/jackson/datatype) | minor | `2.13.4` -> `2.14.0` |
| [com.fasterxml.jackson.module:jackson-module-parameter-names](https://nexus.ocado.tech/service/rest/repository/browse/maven-central/com/fasterxml/jackson/module) | minor | `2.13.4` -> `2.14.0` |
| [com.fasterxml.jackson.module:jackson-module-scala](https://nexus.ocado.tech/service/rest/repository/browse/maven-central/com/fasterxml/jackson/module) | minor | `2.13.4` -> `2.14.0` |

---

**Update diff**

![image](https://user-images.githubusercontent.com/19594920/202139573-aa97b49b-c60f-433f-b84a-b0aa246f305d.png)

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by Elm0xz** [comment]
Yes - all packages mentioned in MR use `jacksonVersion` variable, only `jackson-databind` (that wasn't mentioned as a package to be updated by this MR) uses `jacksonDatabindVersion`. Actually, there was another parallel MR for `jackson-databind` opened by Renovate, where it got properly updated to latest available version that is `2.14.0`.

**Comment by rarkins** [comment]
So you have multiple packages using the same variable, but they don't all have the same latest release?

**Comment by Elm0xz** [comment]
No. All packages using `jacksonVersion` are using the same version and that's working as intended. That's not the crux of the issue here - the issue is that `jacksonDatabindVersion` got updated by MR that had nothing to do with it.

**Comment by rarkins** [comment]
OK, thanks for clarifying. In that case we definitely need a reproduction repo

**Comment by Elm0xz** [comment]
OK, I will try to provide you with reproduction repo

**Comment by Elm0xz** [comment]
Here is the repo with [reproduced issue](https://github.com/Elm0xz/renovate-issue/pull/5) - renovate tries to update `jacksonDatabind` to version that [does not exist](https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind). Also you can see this dependency is not even mentioned in a list of modified dependencies (in fact, Renovate tried to update it earlier independently - you can see it in another MR [here](https://github.com/Elm0xz/renovate-issue/pull/2).

We debugged the Renovate code independently and it seems `auto-replace.ts` could be at fault here

**Comment by rarkins** [comment]
Reproduction forked to https://github.com/renovate-reproductions/18937

It seems that string is found twice and autoreplaced twice
